### ISSUE TITLE
LI-RAGE: Late Interaction Retrieval Augmented Generation with Explicit Signals for Open-Domain Table Question Answering

### ISSUE THREAD

**Issue by bigshanedogg** [bigshanedogg/survey#33]
## Problem statement
1. table은 대체로 매우 크고, row/column 별로 정보가 이질적이기 때문에, reader가 질문에 따라 사용해야 하는 정보가 유동적일 수 있기 때문에 fine-grained Interaction이 필요하다.
  - ODQA에서 자주 사용되는 retriever 중 Dense Passage Retriever 류는 fixed embeddings을 전제로 한다.
2. RAG (혹은 T-RAG)의 세팅에서 가져오는 retriever-reader 구조는 implicit signal만을 사용함
  - retriever의 결과가 동작했는지가 retriever에 업데이트되지 않음. -> relevance token & contrastive loss
  
## Glossary
- ODQA (Open-domain QA): 다양한 주제에 대한 대량의 문서 집합으로부터 자연어 질의에 대한 답변을 찾아오는 태스크
  - 문서 집합에서 참조할 정보를 추출하는 retriever와 추출된 정보로부터 답을 생성하는 reader가 주된 모듈
  - 일반적인 구조:
    - A) retriever-only
    - B) retriever-reader
    - C) generator-only

## Baseline
- 기존 접근법들:
  - ColBERT: DPR의 embeddings를 token-level representation으로 만들어 Latent Interaction을 하는 경우
    - <img width="437" alt="Screen Shot 2023-10-08 at 10 23 57 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/a2cc90e6-b0af-4374-909e-9725cd5ddaaf">
  - T-RAG (Table Retrieval-augmented Generation): retrieval passages로 table을 row/column 단위로 decomposing한 passages를 전달하여 RAG
    - RAG: <img width="618" alt="Screen Shot 2023-10-08 at 10 21 55 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/9aec0d45-efd2-4d45-a350-4eb11d9aeba4">
    - T-RAG: <img width="1091" alt="Screen Shot 2023-10-08 at 10 23 04 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/a8f394a5-653f-41a6-93cb-02cfec0179e7">
    - ANN (Approximate Nearest Neighbor)

## Data details
|name|abbr|type|format|source|size|description|remark|related tasks|
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
|Natural Questions Tables|NQ-TABLE|text||Wikipedia||||open-domain table question answering|
|E2E WikiTableQuestions|E2E-WTQ|text||Wikipedia||||open-domain table question answering|

## Approach
- 기본적으로 Retriever-Reader 구조를 사용하되, table-retriever가 reader와 fine-grained interaction을 하도록 하는 것에 포커스
- table-retriever 시 Query의 각 token과 table의 각 token 간의 fine-grained similarity를 기반으로 (c.f. FiLIP)
  - 이 때, DPR 기반 similarity는 값이 가지는 의미가 적기 때문에, top-k table을 반환함
  - latent interaction: <img width="350" alt="Screen Shot 2023-10-08 at 10 32 04 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/69030f78-05d5-4a7e-b410-1f0b0f7992d5">
  - contrastive learning: <img width="460" alt="Screen Shot 2023-10-08 at 10 34 15 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/95f0f47d-13f9-4624-bdfb-b6cc69ce9c7a">
- retrieval-based Table QA
  - retrieved information을 모두 reader에 전달하는 RAG와 다르게, k개 각각을 입력으로 받았을 때의 predictions 중 가장 logit이 높은 결과를 최종 prediction으로 사용함
  - <img width="266" alt="Screen Shot 2023-10-08 at 10 45 19 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/7914211c-63c5-4b07-a55f-a34f56b29ca7">
- guiding relevance to reader
  - 생각보다 심플한 아이디어인데, QA로 answer 생성 전에 'yes' 혹은 'no'를 생성한 후에 answer를 생성하도록 함.
    - 입력으로 받은 table embedding이 관련이 있으면 yes, 관련이 없으면 no
- retriever-reader 간의 joint training
  - table이 gold answer를 포함하고 있지 않거나 question-answering에 도움이 되지 않을 경우, penalize하기 위함
  - <img width="436" alt="Screen Shot 2023-10-08 at 10 59 49 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/a80c857c-ed1d-434c-86f4-0701b786d152">
  - (앞의 항은 generation loss) 뒤의 항이 retriever의 top-k 결과물에 대한 in-batch contrastive loss
    - c.f. generation loss만 있으면 RAG loss, 뒤의 retriever를 업데이트하는 항이 추가되면 RAGE loss라고 줄여부르는 것 같음
  - 이 때, k개 중 generation 결과에 answer가 포함이 된 경우를 positive로 pseudo-labeling
- 학습 파이프라인
  - Retriever: DPR, ColBERT를 타겟 데이터셋에 대해 finetune 후, reader와 함께 joint training
  - Reader: from scratch가 아닌 TaPEx (BART 기반의 현재 sota Tabular Understanding 모델)에서 시작

## Evaluation
- Retriever 평가
  - <img width="781" alt="Screen Shot 2023-10-08 at 10 56 15 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/04fb7a0d-f09c-4ec3-8e8a-80d5bb3e3643">
    - query-token 간의 similarity로 retrieval하는 LI가 joint-training 여부와 관계 없이 DPR-based Retrieval보다 높은 성능을 보인다.
    - 또한, 2가지 세팅에서 모두 joint training 시에 성능이 더 큰 폭으로 개선됨을 알 수 있음
- QA 평가
  - <img width="904" alt="Screen Shot 2023-10-08 at 10 58 03 AM" src="https://github.com/bigshanedogg/survey/assets/18084680/261a9c7d-81d7-4430-9dd4-132d1dcb7da7">
  - retriever-level에서 뿐만 아니라, QA-level에서도 joint training의 효과를 확인할 수 있음
  - relevance token의 효과도 확인할 수 있음

## Limitations 
- 저자가 언급한 한계:
  - 평가한 benchmark가 2개로 부족함
  - 현재는 BART의 decoder length를 1024로 설정했는데, table 정보의 특성상 이 sequence로는 부족한 경우가 있을 수 있음
